packages/@n8n/task-runner/src/js-task-runner/js-task-runner.ts

```diff
@@ -255,10 +282,26 @@ export class JsTaskRunner extends TaskRunner {
 			};
 
 			try {
-				let result = (await runInNewContext(
-					`module.exports = async function VmCodeWrapper() {${settings.code}\n}()`,
-					context,
-				)) as INodeExecutionData | undefined;
+				let result = await new Promise<INodeExecutionData | undefined>((resolve, reject) => {
+					signal.addEventListener(
```

Comment:
Since this runs in a loop, we are adding a new event listener on each iteration, and it could be a thousand of these. I think it would be good in this case to remove the event listener once we are done. Also then we don't get these warnings:

```
2024/12/10 10:09:30 ERROR [Runner] (node:28928) MaxListenersExceededWarning: Possible EventTarget memory leak detected. 11 abort listeners added to [AbortSignal]. MaxListeners is 10. Use events.setMaxListeners() to increase limit
```

Reply:
Interesting, I thought this was covered by `{ once: true }`